native/c/test/zutils.cpp

```diff
@@ -118,7 +118,9 @@ string get_user()
   if (s_user.empty())
   {
     string user;
-    execute_command_with_output("whoami", user);
+    // Note: using `basename $HOME` instead of `whoami` to get the current user
+    // because `whoami` may be mapped to a kernel user instead of a real one.
+    execute_command_with_output("basename $HOME | tr '[:lower:]' '[:upper:]'", user);
```

Comment:
`getenv("USER")` should also work - it saves the overhead of executing a command & transforming output as its just a C API.